flake.nix

```diff
@@ -0,0 +1,83 @@
+# SPDX-FileCopyrightText: 2022 Oxhead Alpha
+# SPDX-License-Identifier: LicenseRef-MIT-OA
+
+{
+  description = "The tezos-packaging flake";
+
+  inputs = {
+    nixpkgs.url = "github:serokell/nixpkgs?rev=be220b2dc47092c1e739bf6aaf630f29e71fe1c4";
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    serokell-nix = {
+      url = "github:serokell/serokell.nix?rev=46d762e5107d10ad409295a7f668939c21cc048d";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      inputs.nixpkgs.follows = "nixpkgs";
+      flake = false;
+    };
+    opam-nix = {
+      url = "github:tweag/opam-nix?rev=44b97c71a379bb9517135600c7d5e83d941e65d9";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    opam-repository = {
+      url = "git+https://gitlab.com/tezos/opam-repository?rev=845375388d477153b36344f69a23a1b34b2bf82e";
+      flake = false;
+    };
+    tezos = {
+      url = "git+https://gitlab.com/tezos/tezos?rev=cb9f439e58c761e76ade589d1cdbd2abb737dc68";
+      flake = false;
+    };
+  };
```

Comment:
1. No need to specify commit hashes here
2. There's a `gitlab:` schema for gitlab, e.g. `gitlab:tezos/tezos`
3. You can use [serokell's flake-registry](https://github.com/serokell/flake-registry) to reduce the number of inputs here